SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs

```diff
@@ -1,65 +1,223 @@
-        }
-
-        [TestMethod]
-        public void GetGiftForUser_ReturnsUsersFromService()
-        {
-            var gift = new Gift
-            {
-                Id = 3,
-                Title = "Gift Tile",
-                Description = "Gift Description",
-                Url = "http://www.gift.url",
-                OrderOfImportance = 1
-            };
-            var testService = new TestableGiftService
-            {
-                ToReturn =  new List<Gift>
-                {
-                    gift
-                }
-            };
-            var controller = new GiftController(testService);
-
-            ActionResult<List<DTO.Gift>> result = controller.GetGiftForUser(4);
-
-            Assert.AreEqual(4, testService.GetGiftsForUser_UserId);
-            DTO.Gift resultGift = result.Value.Single();
-            Assert.AreEqual(gift.Id, resultGift.Id);
-            Assert.AreEqual(gift.Title, resultGift.Title);
-            Assert.AreEqual(gift.Description, resultGift.Description);
-            Assert.AreEqual(gift.Url, resultGift.Url);
-            Assert.AreEqual(gift.OrderOfImportance, resultGift.OrderOfImportance);
-        }
-
-        [TestMethod]
-        public void GetGiftForUser_RequiresPositiveUserId()
-        {
-            var testService = new TestableGiftService();
-            var controller = new GiftController(testService);
-
-            ActionResult<List<DTO.Gift>> result = controller.GetGiftForUser(-1);
-            
-            Assert.IsTrue(result.Result is NotFoundResult);
-            //This check ensures that the service was not called
-            Assert.AreEqual(0, testService.GetGiftsForUser_UserId);
-        }
-    }
-}
+using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SecretSanta.Api.Controllers;
+using SecretSanta.Domain.Models;
+using System.Collections.Generic;
+using System.Linq;
+using Moq.AutoMock;
+using SecretSanta.Domain.Services;
+using Moq;
+
+namespace SecretSanta.Api.Tests
+{
+    [TestClass]
+    public class GiftControllerTests
+    {
+        private AutoMocker Mocker { get; set; }
+        private Mock<IGiftService> GiftServiceMock { get; set; }
+        private Mock<GiftController> GiftControllerMock { get; set; }
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            Mocker = new AutoMocker();
+            GiftServiceMock = Mocker.GetMock<IGiftService>();
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void GiftController_RequiresGiftService()
+        {
+            new GiftController(null);
+        }
+
+        [TestMethod]
+        public void GetGiftForUser_ReturnsGiftsFromService()
+        {
+            var mocker = new AutoMocker(MockBehavior.Strict);
+            var service = mocker.GetMock<IGiftService>();
+            var controller = new GiftController(service.Object);

```

Comment:
Part of the advantage of using AutoMocker is not needing to invoke the controller's constructor directly (which makes you manually specify all of the constructor parameters). If you let AtuoMocker create the instances it will populate the constructor parameters for you, making your test more resilient to constructor signature changes.
```suggestion
            var controller = mocker.CreateInatance<GiftController>();
```